Source/Core/InputCommon/ControllerEmu/ControlGroup/IMUCursor.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <chrono>
+#include <string>
+
+#include "InputCommon/ControllerEmu/StickGate.h"
+#include "InputCommon/ControllerInterface/Device.h"
+
+namespace ControllerEmu
+{
+class IMUCursor : public ControlGroup
+{
+public:
+
+  IMUCursor(std::string name, std::string ui_name);
+
+  // Yaw movement in radians.
+  ControlState GetTotalYaw() const;
```

Comment:
Why does this setting need to exist? To reach the screen edges can't the user just tilt less/more as required for each game just like they would have to with a physical Wii Remote?

Reply:
As explained in the PR description (under Pointer emulation), this is to remedy motion sensor drift. It's an alternative to the recenter functionality.

Reply:
I understand the issue of drift and recentering. But when would you ever set this to something other than 180 degrees? You seem to have not needed TotalPitch so surely TotalYaw is not needed either?

Reply:
When playing Super Mario Galaxy 2 (with the star munching pointer) I felt this function was handy, just turning the controller a bit left or right and back to correct drift and have a comfortable hand position in my lap. Limiting yaw to the screen edges makes this action more predictable/intuitive and reduces the amount of movement you need.

The reason for not needing TotalPitch is because that is compensated for by combining the (noisy but non-drifting) accelerometer provided pitch an roll angles with the (non-noisy but drifting)  gyro angles. Yaw is however only provided by the gyro.

Reply:
I edited my previous comment to clarify the explanation of why yaw drifts but not pitch. Please check it out if it made no sense earlier.

An advantage to having the TotalYaw setting as an alternative to a recenter button is simply that you free up a button on the controller.

Anyway, I guess a decision must be made. Drop or keep?

Reply:
I understand why there is yaw drift, but I don't know how "TotalYaw" helps with that or how it's an alternative to a recenter button. It's just a scalar, essentially, yeah?

Reply:
Do you mean scalar as in a single number and not for example a 3D vector?  If so, yes.

I'm struggling with coming up with a good abstract explanation for how TotalYaw counters drift, so I'll give a concrete example of what happens when in use.

Imagine sitting in a comfy chair in front of your TV playing Super Mario Galaxy 2 on Dolphin using a DualShock 4 controller. After a while you notice that the controller must be held at an angle horizontally (yawed) to have the on screen pointer centered. This is caused by gyro drift.

To counter this with a recenter button, you point the controller (**not** the on screen pointer) to the center of the screen and press the button mapped to Recenter.

To counter using TotalYaw: Assume TotalYaw is set to match the left and right screen edges. 15 degrees seems good for Super Mario Galaxy 2. Say the pointer is a bit off to the right on the screen when the controller is pointing to the center of the screen. Yaw the controller to the right until the pointer hits the edge of the screen. Now yaw the controller to the right a bit further. Then move the controller back to point at the center of the screen. If the amount of extra controller movement (after the pointer hit the screen edge) was correct, pointer and controller positions will now match up again.

Reply:
Oh, you clamp the rotational velocities to not simulate movement beyond the "TotalYaw" setting. Like a mouse controlling a cursor clamped within the screen? I get it then.

Reply:
Exactly! That's the exact analogy I used in the PR description :-)

Reply:
I admit this may be an unusual way to counter gyro drift, and that it is a bit difficult to explain. However, testing with Super Mario Galaxy 2 I felt it was intuitive and I quickly and without much effort learned to judge how much extra to yaw the controller to resynchronize with the on-screen pointer. I am only one data point however, and it may turn out that no-one else will like this feature. Should I let it stay for now?

Reply:
Seems fine.